### ISSUE TITLE
Allow other columns than assignment registers to be passed to an operation

### ISSUE THREAD

**Issue by georgwiese** [powdr-labs/powdr#1135]
**Example**: In #1129, I implemented a memory machine with operations like
```
operation mload<0> m_addr, m_step -> m_value;
```

which are currently used in the main machine like this:
```
instr mload X, Y -> Z ~ memory.mload;
```

Like this, the current time step has to be passed explicitly. It should instead be implicit, like so:
```
col fixed STEP(i) { i };
instr mload X, STEP -> Z ~ memory.mload;
```

(Currently fails with `All lhs params must be assignment registers`)

Which should compile to:
```
main.instr_mload { 0, main.X, main.STEP, main.Z } is main_memory.selectors[0] { main_memory.m_is_write, main_memory.m_addr, main_memory.m_step, main_memory.m_value };
```

**Comment by pacheco** [comment]
should work as follows: 
`instr mload X -> Z  ~ memory.mload X, STEP -> Z`

**Comment by georgwiese** [comment]
Sadly, I get the same error!

**Comment by pacheco** [comment]
Ahhh right, this is currently not supported: all arguments must be registers, and STEP is a fixed colum.
But indeed i think it should be allowed, maybe even literals should be usable directly here.
But the error should be different here: `All rhs params must be registers`, that's the error you get right?

As a temporary workaround, can change it to `reg STEP;` and then just assigning it to `1`?

**Comment by georgwiese** [comment]
Yes, that works! I think it should allow any expression, including literals and columns.

**Comment by pacheco** [comment]
implemented in #1165
